geopandas/plotting.py

```diff
@@ -91,17 +91,26 @@ def plot_polygon_collection(
         )
     from matplotlib.collections import PatchCollection
 
-    geoms, values = _flatten_multi_geoms(geoms, values)
-    if None in values:
-        values = None
+    geoms, multiindex = _flatten_multi_geoms(geoms, range(len(geoms)))
+    if values is not None:
+        values = [values[i] for i in multiindex]
+        if None in values:
+            values = None
```

Comment:
I think this second check is not needed anymore? 

Before, it was only done because `values` was overwritten by flatten_multi_geoms, in which we did:

```
   if colors is None:
        colors = [None] * len(geoms)
```

but now we know if values was None or not

Reply:
and same for occurence below

Reply:
I actually removed it at one point, but then returned back. You can do `plot_point_collection(ax, points, [0, None, 3])`, and this None is not captured by the `if values is not None` and would raise an error. if` plot_??_collection` functions were private (#656) we could remove it as it actually raises an error before if passed to plot_geodataframe. In this situation its removal might cause errors.

Reply:
Is there a test case relying on that? 
I would say that passing such values is just a user error, and it is not a problem that this raises an error in matplotlib. Also now, by the `if None in values` check, we completely ignore all the other valid values just because there is a single None in it, which also seems strange behaviour ?

Reply:
> Is there a test case relying on that?

II have added tests to check it as it was previously tested together with pure None (or rather not tested by accidentally covered). 

I completely agree with the rest, but I just didn't want to change the behaviour as this is essentially only bug fix. What should be done though, is to check for None and raise error, here as well as in `plot_dataframe`, or (better) support plotting with missing data in the same way as QGIS does (row is ignored). That would cover these cases, but that should be separate PR (which I am happy to do later).